src/internals/scan.rs

```diff
@@ -252,20 +263,21 @@ pub fn rules_scan_proc<'a>(
 pub fn scanner_scan_proc<'a>(
     scanner: *mut yara_sys::YR_SCANNER,
     pid: u32,
-) -> Result<Vec<Rule<'a>>, YaraError> {
-    let mut results = Vec::<Rule<'a>>::new();
+    callback: impl FnMut(CallbackMsg<'a>) -> CallbackReturn,
+) -> Result<(), YaraError> {
+    let p_callback = Box::new(Box::new(callback));
```

Comment:
Why the double box? I think you can just pass the pointer to the inner box to `yr_scanner_set_callback`, something like:
```
let boxed_callback = Box::new(callback);
yara_sys::yr_scanner_set_callback(
    // ...
    &p_callback as *mut _,
);
```

In addition, you have a memory leak when you call `Box::into_raw`.

It also might be possible to directly pass the callback pointer, like this, but I'm not sure:
```
&callback as *mut _,
```

Reply:
I read about this here: https://stackoverflow.com/questions/32270030/how-do-i-convert-a-rust-closure-to-a-c-style-callback

Fix memory leak.

Reply:
@Hugal31 maybe you can help me with Windows? I never work on this platfomr